lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -303,6 +304,29 @@ def rebuild_secrets(inventory, input)
 
         inventory
       end
+
+      def prune_known_hosts_on_first_execution(hosts)
+        hosts.each do |value|
+          action_input = value[:input][:action_input]
+          next unless action_input.fetch(:first_execution, false)
+
+          hostname = action_input.fetch(:name)
+          host_ip = action_input.fetch(:hostname)
```

Comment:
How does a hostname turn into an IP?

Reply:
Here's how the IP is being sent in the JSON.